packages/std/src/math.rs

```diff
@@ -371,15 +436,15 @@ impl<'de> de::Visitor<'de> for Uint128Visitor {
     }
 }
 
-impl Sum<Uint128> for Uint128 {
-    fn sum<I: Iterator<Item = Uint128>>(iter: I) -> Self {
-        iter.fold(Uint128::zero(), ops::Add::add)
+impl Sum<Uint128> for Result<Uint128, OverflowError> {
```

Comment:
Also seems more unwieldy to me. I agree with allowing opt-in, but why as default?
It makes the whole API more complex and doesn't give more security in this extreme edge case (contract panic will halt), just a better error message.

Reply:
It makes `+` and `-` interface more consistent, also make user aware of the overflow error case.

Reply:
I see it as a case that is never hit in normal usage and impedes the usage experience for normal user.

If I am doing math for a bonding curve and squaring amounts and adding them, it is very important to use `checked_add`, and great to expose them as you do in this PR. But in this case, I assume the developer will use the methods not `+` (or quickly realize that on the first tests that overflows and panics)

Reply:
API changes ar @webmaster128 judgement, I just find it to be a bit more annoying to enforce checked add constantly. But if you both prefer it, I will use it.